Server/ReasnAPI/ReasnAPI/Services/InterestService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+
+namespace ReasnAPI.Services;
+
+public class InterestService(ReasnContext context)
+{
+    public InterestDto CreateInterest(InterestDto interestDto)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Name == interestDto.Name);
+        if (interest != null)
+        {
+            return null;
+        }
+
+        var newInterest = new Interest
+        {
+            Name = interestDto.Name
+        };
+
+        context.Interests.Add(newInterest);
+        context.SaveChanges();
+        return interestDto;
+    }
+
+    public InterestDto UpdateInterest(int interestId, InterestDto interestDto)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Id == interestId);
+        if (interest == null)
+        {
+            return null;
+        }
+
+        interest.Name = interestDto.Name;
+
+        context.Interests.Update(interest);
+        context.SaveChanges();
+        return interestDto;
+    }
+    public bool DeleteInterest(int id)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Id == id);
+
+        if (interest == null)
+        {
+            return false;
+        }
+        context.Interests.Remove(interest);
+        context.SaveChanges();
+
+        return true;
+    }
```

Comment:
What if the interest is being used by some user? Do we still want to delete it?

Reply:
I added a check to see if it's used. If it is, the method will return false.